eng/scripts/Generate-AOTCompactReport.ps1

```diff
@@ -0,0 +1,195 @@
+#!/bin/env pwsh
+#Requires -Version 7
+
+param(
+    [string]$Runtime,
+    [bool]$GenerateHtml = $true
+)
+
+$ErrorActionPreference = "Stop"
+
+. "$PSScriptRoot/../common/scripts/common.ps1"
+$root = $RepoRoot.Path.Replace('\', '/')
+$projectFile = "$root/src/AzureMcp.csproj"
+# Base directory for all AOT compatibility reports
+$aotReportDir = "$root/.work/aotCompactReport"
+# the file containing the raw AOT compact output generated by the `dotnet publish` command.
+$reportPath = "$aotReportDir/aot-compact-report.txt"
+# A JSON file where each entry is a DLL name and AOT warnings from types in that DLL.
+$jsonReportPath = "$aotReportDir/aot-compact-report.json"
+
+if (-not $Runtime) {
+    $runtime = [System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier
+} else {
+    $runtime = $Runtime
+}
+
+function Initialize-AOTAnalysisDirectories {
+    param(
+        [string]$aotReportDir,
+        [string]$projectFile
+    )
+    
+    if (!(Test-Path $aotReportDir)) {
+        # Ensure AOT report directory exists
+        New-Item -ItemType Directory -Path $aotReportDir -Force | Out-Null
+    }
+    
+    $projectObjDir = Join-Path (Split-Path $projectFile) "obj"
+    if (Test-Path $projectObjDir) {
+        # Clean project obj directory to ensure fresh build
+        Write-Host "Deleting project obj directory: $projectObjDir"
+        Remove-Item -Path $projectObjDir -Recurse -Force -ErrorAction SilentlyContinue
+    }
+}
+
+function Get-DllNameSet {
+    param(
+        [string]$projectFile
+    )
+    $objDir = Join-Path (Split-Path $projectFile) "bin"
+    $dllNameTable = @{}
+    if (Test-Path $objDir) {
+        $dlls = Get-ChildItem -Path $objDir -Recurse -Filter *.dll -File
+        foreach ($dll in $dlls) {
+            $dllName = $dll.BaseName
+            $dllNameTable[$dllName] = $true
+        }
+    }
+    # Sort: Returns DLL names sorted with the most specialized (most dots) first.
+    return $dllNameTable.Keys |
+        Sort-Object @{ Expression = { $_.Split('.').Count }; Descending = $true },
+                    @{ Expression = { $_ }; Descending = $true }
+}
+
+function Extract-TypeNameFromILWarningLine {
+    param(
+        [string]$line
+    )
+    # Match e.g. 'warning IL2026: Azure.Monitor.OpenTelemetry.AspNetCore.DefaultAzureMonitorOptions.Configure(AzureMonitorOptions): ...'
+    if ($line -match 'warning IL\d+: ([^:]+):') {
+        $full = $Matches[1]
+        # Extract type name before last dot and first parenthesis
+        if ($full -match '^(.*)\.([^.]+)\(') {
+            return $Matches[1]
+        } else {
+            return $full
+        }
+    }
+    return $null
+}
+
+function Find-BestDllMatch {
+    param(
+        [string]$typeName,
+        [string[]]$dllNames # List of DLL names sorted with most specialized first
+    )
+    
+    # Only look for exact matches or prefix matches i.e., typename starts with DLLName
+    # This is the most reliable heuristics for .NET assemblies.
```

Comment:
This heuristic is our best effort, and it is naming pattern followed by our Azure .NET libraries as well as many other out there. It works well for types, if non-matching types found we categorize those under a common section (unknown). This helps to produce a good json and html report. The raw report is always there.